### PR TITLE
2677 - Add sendgrid api key to deploy-backend

### LINKED ISSUE #2677: add `SENDGRID_API_KEY` to deploy.backend.sh
https://github.com/raft-tech/TANF-app/blob/a980d1c2ed0466f6371b072e1f8d0515bfa4414f/scripts/deploy-backend.sh#L55

currently the env var in circleci is not being supplied to the backend apps on hhs side. 

Acceptance criteria:
- [ ] add `SENDGRID_API_KEY` to `deploy-backend.sh` to be managed by circleci
- [ ] document that all environment variables should be managed by circleci

### PR DESCRIPTION
**Pull request by jtimpe**
## Summary of Changes
Pull request closes #2677 

## How to Test

1. In the TDRS Project Settings in circleci, add a new Environment Variable: `SENDGRID_API_KEY` (or verify/modify the existing value). If you change it, you'll be able to see the change in the final step
1. Trigger a circleci deployment to a cloud.gov environment of your choice
1. In the `tdrs-backend-{env}` environment variables in cloud.gov, verify the value from circleci is in place for `SENDGRID_API_KEY`

Proof from my test
image

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] add `SENDGRID_API_KEY` to `deploy-backend.sh` to be managed by circleci
+ [x] document that all environment variables should be managed by circleci
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by elipe17 — APPROVED** [review]
Checked CirCI and the Raft env and verified the var was there.

**Review by raftmsohani — APPROVED** [review]
LGTM!

File: scripts/deploy-backend.sh [hunk 1]
```diff
@@ -62,6 +62,7 @@ set_cf_envs()
   "REDIS_URI"
   "JWT_KEY"
   "STAGING_JWT_KEY"
+  "SENDGRID_API_KEY"
```
Comment (jtimpe) [inline scripts/deploy-backend.sh:65]: document in security controls

**Review by ADPennington — APPROVED** [review]
backend before deployment
sendgridpr1

backend after deployment
sendgridpr2

email sent ✔️ 
sendgridpr3

**Pull request merged**